csharp/extractor/Semmle.Extraction.CSharp.Standalone/AssemblyInfo.cs

```diff
@@ -101,26 +77,74 @@ public IEnumerable<string> IndexStrings
         }
 
         /// <summary>
-        /// Get an invalid assembly info (Valid==false).
+        /// Get an invalid assembly info.
         /// </summary>
         public static AssemblyInfo Invalid { get; } = new AssemblyInfo();
 
-        private AssemblyInfo() { }
+        // Constructs a partial AssemblyInfo for the above Invalid instance.
+        private AssemblyInfo()
+        {
+            // Initializing non null fields.
+            Name = string.Empty;
+            Filename = string.Empty;
+        }
+
+        // Constructs only a partial AssemblyInfo, FileName is not specified
+        private AssemblyInfo(string id) : this()
+        {
+            var sections = id.Split(new string[] { ", " }, StringSplitOptions.None);
+
+            Name = sections.First();
+
+            foreach (var section in sections.Skip(1))
+            {
+                if (section.StartsWith("Version="))
+                    Version = new Version(section.Substring(8));
+                else if (section.StartsWith("Culture="))
+                    Culture = section.Substring(8);
+                else if (section.StartsWith("PublicKeyToken="))
+                    PublicKeyToken = section.Substring(15);
+                // else: Some other field like processorArchitecture - ignore.
+            }
+        }
+
+        private AssemblyInfo(string id, string filename) : this(id)
+        {
+            Filename = filename;
+        }
+
+        private AssemblyInfo(string filename, string name, Version version, string culture, string publicKeyToken)
+        {
+            Filename = filename;
+            Name = name;
+            Version = version;
+            Culture = culture;
+            PublicKeyToken = publicKeyToken;
+        }
 
         /// <summary>
         /// Get AssemblyInfo from a loaded Assembly.
         /// </summary>
         /// <param name="assembly">The assembly.</param>
         /// <returns>Info about the assembly.</returns>
-        public static AssemblyInfo MakeFromAssembly(Assembly assembly) => new AssemblyInfo() { Valid = true, Filename = assembly.Location, Id = assembly.FullName };
+        public static AssemblyInfo MakeFromAssembly(Assembly assembly)
+        {
+            if (assembly.FullName is null)
```

Comment:
this is a slight change in functionality. `FullName` can be `null`, now we're explicitly throwing an exception, previously the first access would have thrown (a different type of exception).